comit/src/expiries.rs

```diff
@@ -0,0 +1,591 @@
+
+        match next_action {
+            WaitForAlphaFunded => {
+                // Transition from Created to AlphaFunded
+                c.alice_to_act() + c.include_alpha_fund_transaction() + c.finality_alpha()
+            }
+            FundBeta => {
+                // Transition from AlphaFunded to FundBetaTransactionSent
+                c.bob_to_act()
+            }
+            WaitForBetaFundTransactionFinality => {
+                // Transition from FundBetaTransactionSent to BetaFunded
+                c.include_beta_fund_transaction() + c.finality_beta()
+            }
+            WaitForBetaRedeemTransactionToBeSeen => {
+                // Transition from BetaFunded to RedeemBetaTransactionSeen
+                c.alice_to_act() + c.include_beta_redeem_transaction()
+            }
+            RedeemAlpha => {
+                // Transition from RedeemBetaTransactionSeen to
+                // RedeemAlphaTransactionSent
+                c.bob_to_act()
+            }
+            WaitForAlphaRedeemTransactionFinality => {
+                // Transition from RedeemAlphaTransactionSent to Done
+                c.include_alpha_redeem_transaction() + c.finality_alpha()
+            }
+            // Transitioning to any of the cancel actions takes, be definition, zero time.
+            NoFurtherAction | Abort | WaitForRefund | Refund => Duration::zero(),
+        }
+    }
+
+    /// True if Bob has funded the beta HTLC. We return true as soon as the
+    /// fund transaction has been sent since any time after attempting to refund
+    /// is the correct cancellation action.
+    fn has_funded_beta(&self) -> bool {
+        match self {
+            BobState::Created | BobState::AlphaFunded => false,
+            BobState::FundBetaTransactionSent
+            | BobState::BetaFunded
+            | BobState::RedeemBetaTransactionSeen
+            | BobState::RedeemAlphaTransactionSent
+            | BobState::Done => true,
+        }
+    }
+}
+
+// TODO: Add support for lightning.
+#[derive(Clone, Copy, Debug)]
+pub enum Protocol {
+    Herc20Hbit,
+    HbitHerc20,
+}
+
+impl Protocol {
+    fn config(&self) -> Config {
+        match self {
+            Protocol::Herc20Hbit => Config {
+                alpha_required_confirmations: ethereum_required_confirmations(),
+                beta_required_confirmations: bitcoin_required_confirmations(),
+            },
+            Protocol::HbitHerc20 => Config {
+                alpha_required_confirmations: bitcoin_required_confirmations(),
+                beta_required_confirmations: ethereum_required_confirmations(),
+            },
+        }
+    }
+}
+
+/// Configuration values used during transition period calculations.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Config {
+    alpha_required_confirmations: usize,
+    beta_required_confirmations: usize,
+}
+
+// TODO: Calculate _real_ values instead of just returning 1 hour.
```

Comment:
Wasn't this done as part of the spike?

Reply:
No, that is the reason the spike is not merged :)